File: examples/erlang/weather-service/src/weather_service_impl.erl
-module(weather_service_impl).

%% This file will NOT be overwritten. Add your business logic here.
-behaviour(weather_service_handler).

-export([
    create_weather_report/2,
    get_weather/2
]).

-spec create_weather_report(weather_service_handler:create_weather_report_input(), map()) ->
    {ok, weather_service_handler:create_weather_report_output()} | {error, term()}.
create_weather_report(_Input, _Context) ->
    {error, not_implemented}.

-spec get_weather(weather_service_handler:get_weather_input(), map()) ->
    {ok, weather_service_handler:get_weather_output()} | {error, term()}.
get_weather(_Input, _Context) ->
    {error, not_implemented}.

File: examples/elixir/weather-service/lib/weather_service_impl.ex
defmodule WeatherService.Impl do
  @moduledoc false
  @behaviour WeatherService.Handler

  # This file will NOT be overwritten. Add your business logic here.

  @spec get_weather(map(), map()) :: {:ok, map()} | {:error, term()}
  def get_weather(%{city: _city}, _ctx) do
    {:ok, %{"temperature" => 22.5, "unit" => "Celsius"}}
  end

  @spec create_weather_report(map(), map()) :: {:ok, map()} | {:error, term()}
  def create_weather_report(%{city: city, temperature: temp, unit: unit}, _ctx) do
    report_id = "rpt-#{city}-#{temp}-#{unit}"
    {:ok, %{"reportId" => report_id}}
  end
end
